CONTRIBUTING.md

```diff
@@ -86,6 +86,18 @@ expanding into more specifics.
    with a few days. Although, depending on the circumstances, it may take
    longer.
 
+### New sources, sinks, and transforms
+
+If you're contributing a new source, sink, or transform to Vector, thank you that's way cool! There's a couple of things you need think about if you want to make sure we can merge your contribution.
+
+To merge a new source, sink, or transform, you need to:
+
+- [ ] Add tests, especially integration tests if your contribution connects to an external service.
+- [ ] Add instrumentation so folks using your integration can get insight into how it's working and performing. You can see some [example of instrumentation in existing integrations](https://github.com/timberio/vector/tree/master/src/internal_events).
+- [ ] Add internal documentation of options and fields. You can see [examples in the `.meta` directory](https://github.com/timberio/vector/blob/master/.meta/sources/kafka.toml.erb).
+- [ ] Add external documentation explaining the use of the integration. You can see [some examples in our existing documentation](https://github.com/timberio/vector-website/blob/master/docs/reference/sources/kafka.md.erb).
```

Comment:
We can remove this requirement. These pages will be automatically generated from the `.meta` folder.